node/core/runtime-api/src/lib.rs

```diff
@@ -463,6 +463,17 @@ where
 				});
 				metrics.on_request(res.is_ok());
 				res
+			} else if api_version >= 2 {
```

Comment:
As discussed, we should avoid this pattern: https://github.com/paritytech/polkadot/issues/5038 and instead use a helper function that incapsulates version negotiation.
I'm fine with implementing it as a follow-up though.

Reply:
We had a follow-up discussion with @tdimitrov on that. `SessionInfo` is used so many times in so many different subsystems that it would become a pain in the ass to support that in every subsystem, so we decided to leave it as-is for now. I feel I'm missing something, and a better design is possible here but yes, let's leave it for follow-ups. We collected quite a few refactoring ideas during this PR flow; I'll file a separate issue on them.